### DISCUSSION TITLE
✨: SPAハンズオン側へのフィードバック対応

### DISCUSSION THREAD

**Discussion by nagano-ryo** [ws-4020/mobile-app-crib-notes#321] [General]
モバイルアプリハンズオンで発生したToDo更新の不具合について、同様の問題が発生するか確認したうえで、
SPAハンズオン側へのフィードバック対応を行う。

## 結論
SPAハンズオンでも同様の問題（上から連続的にToDoを更新していくと、途中で意図していないToDoが更新される）が発生した。
修正内容もモバイルアプリハンズオンと同様。

## 対象ページ
https://fintan-contents.github.io/spa-restapi-handson/todo/frontend/todo-update/#todoboard%E3%81%AE%E5%AE%9F%E8%A3%85
└ toggleTodoCompletion()

## 修正コード
```
  const toggleTodoCompletion = (id: number) => {
    const target = todos.find((todo) => todo.id === id);
    if (!target) {
      return;
    }
    BackendService.putTodo(id, !target.completed)
      .then((returnedTodo) =>
        setTodos((prevTodos) => {
          return prevTodos.map((todo) => (todo.id === id ? returnedTodo : todo));
        }),
      )
      .catch(() => {});
  };
```

## 参考情報
- 関数型の更新
https://ja.reactjs.org/docs/hooks-reference.html#functional-updates

- React公式ドキュメント
https://ja.reactjs.org/docs/faq-state.html#what-is-the-difference-between-passing-an-object-or-a-function-in-setstate

**Comment by nagano-ryo** [comment]
SPAハンズオンへフィードバック済み。

**Comment by creature-water-valley** [comment]
SPAハンズオンへフィードバック済みのため、モバイルのリポジトリのディスカッションはクローズする
